From 03581d33bda40ddbe6f62404a59203fa96cd6b2b Mon Sep 17 00:00:00 2001
From: Kyle Hsieh <Kyle.Hsieh@quantatw.com>
Date: Fri, 31 Jul 2026 17:17:59 +0800
Subject: [PATCH] bios: Fix D-Bus bindings and transition to HostSPIFlash types

This commit fixes the D-Bus interface bindings in `main.cpp` to properly
listen for the new `IntelHostSPIFlash` and `HostSPIFlash` configuration
types, ensuring the daemon correctly discovers the devices.

It also updates the BIOS software manager and documentation to strictly
use these new types, completely replacing the generic `SPIFlash` and
`IntelSPIFlash`.

Upstream-Status: Submitted [https://gerrit.openbmc.org/c/openbmc/phosphor-bmc-code-mgmt/+/93036]

Change-Id: I30a4b1ee9f83bf38839dcf04eda5964e7c246231
Signed-off-by: Kyle Hsieh <Kyle.Hsieh@quantatw.com>
---
 bios/README.md                 | 2 +-
 bios/bios_software_manager.cpp | 6 ++----
 bios/main.cpp                  | 4 ++--
 3 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/bios/README.md b/bios/README.md
index 708c433..5bfbf9d 100644
--- a/bios/README.md
+++ b/bios/README.md
@@ -27,7 +27,7 @@ xyz.openbmc_project.Configuration.SPIFlash
     "VendorIANA": "6653",
     "CompatibleHardware": "com.tyan.Hardware.S8030.SPI.Host"
   },
-  "Type": "SPIFlash"
+  "Type": "HostSPIFlash"
 }
 ```
 
diff --git a/bios/bios_software_manager.cpp b/bios/bios_software_manager.cpp
index bb6f19b..725f96a 100644
--- a/bios/bios_software_manager.cpp
+++ b/bios/bios_software_manager.cpp
@@ -48,13 +48,11 @@ sdbusplus::async::task<bool> BIOSSoftwareManager::initDevice(
 
     enum FlashTool tool = flashToolNone;
 
-    if (config.configType == "IntelSPIFlash" ||
-        config.configType == "IntelHostSPIFlash")
+    if (config.configType == "IntelHostSPIFlash")
     {
         tool = flashToolFlashrom;
     }
-    else if (config.configType == "SPIFlash" ||
-             config.configType == "HostSPIFlash")
+    else if (config.configType == "HostSPIFlash")
     {
         tool = flashToolFlashcp;
     }
diff --git a/bios/main.cpp b/bios/main.cpp
index abed9bc..1f3cf81 100644
--- a/bios/main.cpp
+++ b/bios/main.cpp
@@ -11,8 +11,8 @@ void run(bool dryRun)
     sdbusplus::async::context ctx;
 
     std::vector<std::string> configIntfs = {
-        "xyz.openbmc_project.Configuration.SPIFlash",
-    };
+        "xyz.openbmc_project.Configuration.IntelHostSPIFlash",
+        "xyz.openbmc_project.Configuration.HostSPIFlash"};
 
     BIOSSoftwareManager spidcu(ctx, dryRun);
 
-- 
2.53.0-Meta

